tests/functional/modules/test_zos_job_submit_func.py

```diff
@@ -529,7 +529,12 @@ def test_job_submit_max_rc(ansible_zos_module, args):
                 #Expecting: - "The job return code 8 was non-zero in the job output, this job has failed"
                 #           - Consider using module zos_job_query to poll for a long running job or
                 #             increase option \\'wait_times_s` to a value greater than 10.",
-                if result.get('duration') >= args["wait_time_s"]:
+                if result.get('duration'):
+                    duration = result.get('duration')
```

Comment:
I don't really recall what comes back from get duration but i doubt its ever a non-zero value which leads me to question the `duration = 0` else stmt, will we mask the second part of this test by setting it to 0? 

Here:  assert re.search(r'non-zero', repr(result.get("msg")))

Comes down to that if `duration = result.get('duration')` can return non-zero values which I don't think it does, we might need the duration to remain undefined so it can hit the non-zero else noted above.

Reply:
duration is how long (in seconds) a job took to run.  So, if the job runs over, fails before start, or fails to complete, there are several pathways where duration is not returned.

If duration is undefined, the test_zos_job_submit_func routine crashes unit testing, because it is comparing a NonType to an integer, which python doesn't like to do.

I put the value of 0 in there, so that 'duration' is always defined, even if it isn't returned.

Reply:
We went through this today, likely we need to correct zos_job_submit, I see your point and can consider this resolved, thanks @richp405